src/current/v23.1/goldengate.md

```diff
@@ -0,0 +1,458 @@
+
+- Ensure that you have the two required Oracle GoldenGate installations:
+    1. [Oracle Golden Gate for Oracle](https://www.oracle.com/middleware/technologies/goldengate-downloads.htm) is required to pull source data and route it to proper [trail files](https://docs.oracle.com/goldengate/c1230/gg-winux/GGCON/processes-and-terminology.htm).
+    1. [Oracle Golden Gate for PostgreSQL](https://www.oracle.com/middleware/technologies/goldengate-downloads.htm) is required to pull data from the trail files to CockroachDB.
+- For CockroachDB clusters running v22.1 and earlier, enable the following cluster settings:
+
+    ~~~
+    set cluster setting  sql.defaults.datestyle.enabled=true;
+    set cluster setting sql.defaults.intervalstyle.enabled = true;
+    ~~~
+
+    For versions of CockroachDB v22.2 and later, these settings are already enabled by default.
+
+- Ensure you have a secure, publicly available CockroachDB cluster running the latest **{{ page.version.version }}** [production release](https://www.cockroachlabs.com/docs/releases/{{ page.version.version }}), and have created a [SQL user]({% link {{ page.version.version }}/security-reference/authorization.md %}#sql-users).
+
+## Configure Oracle GoldenGate for CockroachDB
+
+This section describes how to configure Oracle GoldenGate to treat CockroachDB as PostgreSQL. Note that this is referring to Oracle GoldenGate for PostgreSQL, the process that pulls data from trail files over to CockroachDB, which is a separate installation from Oracle GoldenGate for Oracle. For more information, refer to the [Oracle GoldenGate for PostgreSQL documentation](https://docs.oracle.com/en/middleware/goldengate/core/19.1/gghdb/preparing-database-oracle-goldengate-postgresql.htm). The examples below will use a [CockroachDB {{ site.data.products.serverless }} cluster](https://cockroachlabs.com/docs/cockroachcloud/create-a-serverless-cluster).
+
+1. Set up the `ODBC.ini` file on Oracle GoldenGate host, and make sure your CockroachDB {{ site.data.products.serverless }} cluster's [root CA certificate](https://cockroachlabs.com/docs/cockroachcloud/connect-to-a-serverless-cluster#connect-to-your-cluster) is in the `TrustStore` path:
+
+    ~~~
+    # This is needed so that all the Postgres libraries can be found
+    export LD_LIBRARY_PATH=/usr/pgsql-13/lib:/u01/ggs-pg/lib
+
+    # This is needed so that OGG knows where to look for connection details for the database
+    export ODBCINI=/etc/odbc.ini
+
+    vi /etc/odbc.ini
+
+    # Inside put this: replace the details for log in with your own. Be sure to prefix the database name with {hostname}.{db}
+    # No changes should be needed for CRDBLOCAL
+    [ODBC Data Sources]
+    PG_src=DataDirect 7.1 PostgreSQL Wire Protocol
+    PG_tgt=DataDirect 7.1 PostgreSQL Wire Protocol
+    CRDB=DataDirect 7.1 PostgreSQL Wire Protocol
+
+    [ODBC]
+    IANAAppCodePage=4
+    InstallDir=/u01/ggs-pg
+
+
+    [CRDBSERVERLESS]
+    Driver=/u01/ggs-pg/lib/GGpsql25.so
```

Comment:
Note that this driver will always point to where your `Oracle Golden Gate for PostgreSQL` installation is and will be in the lib folder and contain the `psql` string.